src/include/index/bwtree.h

```diff
@@ -2864,6 +2866,17 @@ class BwTree : public BwTreeBase {
    * the mapping table rather than CAS with nullptr
    */
   void InitMappingTable() {
+    mapping_table = (std::atomic<const BaseNode *> *) \
+                    mmap(NULL, 1024 * 1024 * 1024, 
+                         PROT_READ | PROT_WRITE, 
+                         MAP_ANONYMOUS | MAP_PRIVATE,
+                         -1, 0);
+    if(mapping_table == (void *)-1) {
```

Comment:
This looks incorrect. If the mmap fails, we log a trace message. Most of the time we don't run with tracing, so the mmap failure is silently ignored. Sometime later, things will fail, but the original failure will not be clear.

Not sure on the correct fix, but:
- LOG_ERROR here, not trace.
- First choice would be to gracefully reject the request to create a BWTree (via an exception or failure of the constructor), OR kill the process via an exception or explicitly.